ui/pages/confirmations/hooks/useTrackERC20WithoutDecimalInformation.ts

```diff
@@ -0,0 +1,58 @@
+import { useSelector } from 'react-redux';
+import { useContext, useEffect } from 'react';
+import { Hex } from '@metamask/utils';
+
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventLocation,
+  MetaMetricsEventName,
+  MetaMetricsEventUiCustomization,
+} from '../../../../shared/constants/metametrics';
+import { TokenStandard } from '../../../../shared/constants/transaction';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import { getCurrentChainId } from '../../../selectors';
+import { parseTokenDetailDecimals, TokenDetailsERC20 } from '../utils/token';
+
+/**
+ * Track event that number of decimals in ERC20 is not obtained
+ *
+ * @param tokenAddress
+ * @param tokenDetails
+ * @param metricLocation
+ */
+const useTrackERC20WithoutDecimalInformation = (
+  tokenAddress: Hex | string | undefined,
+  tokenDetails?: TokenDetailsERC20,
+  metricLocation = MetaMetricsEventLocation.SignatureConfirmation,
+) => {
+  const trackEvent = useContext(MetaMetricsContext);
+  const chainId = useSelector(getCurrentChainId);
```

Comment:
Should we stop relying on this selector, instead can pass a provided `chainId` to hook? `PermitSimulation` component already aware of `chainId` which could be drilled down in to here.

Reply:
We avoid passing down props that can be obtained in child component. Do you see any important reasons to do this ?

Reply:
That's right, it's not ideal but don't we have a goal to make things work in multi-chain? So if we rely on this selector it will just go with selected chain id. 
I think we have whole big epic to clear these use-cases in backlog now (`2024 Q4 - Remove Global Network Usage`)

Reply:
We can change selector in that case, but yep as mentioned there is already an epic planned for this.